### ISSUE TITLE
Add support for determining the key exchange for connection

### ISSUE THREAD

**Issue by filiptronicek** [rustls/rustls#2019]
**Checklist**
* [x] I've searched the issue tracker for similar requests

**Is your feature request related to a problem? Please describe.**

I would like to be able to determine the key exchange algorithm used in a TLS connection.

**Describe the solution you'd like**

I would like to be able to access the key exchange mechanism of a TLS connection, just like I can access the negotiated cipher suite name. It could potentially live somewhere alongside or inside the `.conn.negotiated_cipher_suite()` method.

**Describe alternatives you've considered**

I have tried to set up a `CryptoProvider` with only the desired kex group (`rustls_post_quantum::X25519Kyber768Draft00` in this case), which seems to work on sites including `pq.cloudflareresearch.com`. I have set it up like so:
```rs
let parent = aws_lc_rs::default_provider();
let kyber_provider = CryptoProvider {
    kx_groups: vec![&rustls_post_quantum::X25519Kyber768Draft00],
    ..parent
};

if CryptoProvider::install_default(kyber_provider).is_err() {
    return false;
}
```

The problem with this approach is that it does not work if I added more groups to the `CryptoProvider`, because I would not be able to tell which actually ended up being used. I could only tell that if the handshake succeeded, it had to be one of the ones I provided.

**Comment by djc** [comment]
Why do you want to know this? What's the use case/motivation here? Just research?

**Comment by filiptronicek** [comment]
@djc that's right, I am gathering data on the adoption of key exchange mechanisms. Apart from that, it seems providing some of the data tools like the Google Chrome DevTools show could be useful not just in research, but also things like e2e tests of TLS.
<img width="623" alt="image" src="https://github.com/rustls/rustls/assets/29888641/8f51ac69-e22b-4212-9f95-51dafa5526e3">

**Comment by cpu** [comment]
>  Apart from that, it seems providing some of the data tools like the Google Chrome DevTools show could be useful not just in research, but also things like e2e tests of TLS.

I think providing details from the peer certificate chain is a harder sell but from the "Connection" info shown in your screenshot I think it's just the key exchange information that's missing. From my perspective it does seem reasonable to expose that information so I've drafted a PR in #2028 exposing it.

**Comment by cpu** [comment]
Support for this was added to `main` as [`CommonState::negotiated_key_exchange_group`](https://rustls.github.io/rustls/prerelease/struct.CommonState.html#method.negotiated_key_exchange_group).

**Comment by cpu** [comment]
This is now available as part of [0.23.11](https://github.com/rustls/rustls/releases/tag/v/0.23.11).

**Comment by filiptronicek** [comment]
Awesome to hear! Thank you, @cpu, for the super quick turnaround!
